iproto: override request handlers using event triggers (#9473)
tarantool/tarantool: changelogs/unreleased/gh-8138-iproto-override-via-event-triggers.md

## feature/core

* Now it is possible to set IPROTO request handler callbacks by using the new

>   Lua module `trigger`, e.g.:

Reviewer: ```suggestion
  Lua module `trigger`, for example:
```

Author: Done.

  ```lua
  local trigger = require('trigger')
  trigger.set('box.iproto.override.select', 'my_select', my_select_handler)
  ```

>   This new method works before the initial `box.cfg{}` call, also it allows to
>   set multiple handlers for a single request type (gh-8138).

Reviewer: ```suggestion
  The method works before the initial `box.cfg{}` call. Also, the method allows settinng multiple handlers for a single request type (gh-8138).
```

Author: Done.
